### PR TITLE
Add ogg/vorbis sound for effects

### PR DESCRIPTION
**Pull request by 0lvin**

### CODE REVIEW

**Comment by 0lvin** [comment]
Code is checked with convert all mp3 files to ogg/vorbis in Hardly Workin' from Machinima.

[yquake2: Hardly Workin' - Machinima](http://www.youtube.com/watch?v=mMf1ESxiROM)

File: src/client/sound/ogg.c [hunk 1]
```diff
@@ -709,3 +710,62 @@ OGG_Shutdown(void)
...
+		/* fill in wav structure */
+		info->rate = ogg_file->sample_rate;
+		info->width = 2;
+		info->channels = ogg_file->channels;
+		info->loopstart = -1;
+		info->samples = stb_vorbis_stream_length_in_samples(ogg_file);
+		info->dataofs = 0;
+
+		/* alloc memory for uncompressed wav */
+		final_buffer = Z_Malloc(info->samples * sizeof(short));
```
Comment (DanielGibson) [inline src/client/sound/ogg.c:744]: shouldn't this also take the number of channels into account?

File: src/client/sound/sound.c [hunk 2]
```diff
@@ -205,17 +205,63 @@ S_IsSilencedMuzzleFlash(const wavinfo_t* info, const void* raw_data, const char*
...
+	/* Remove the extension */
+	memset(namewe, 0, 256);
+	memcpy(namewe, path, len - (strlen(ext) + 1));
+
+	if (len < 5)
+	{
+		return;
+	}
+
+	if (strcmp(ext, "wav"))
```
Comment (DanielGibson) [inline src/client/sound/sound.c:239]: what's happening here?

File: src/client/sound/sound.c [hunk 3]
```diff
@@ -205,17 +205,63 @@ S_IsSilencedMuzzleFlash(const wavinfo_t* info, const void* raw_data, const char*
...
+		/* Non wav? */
+		return;
+	}
+
+	/* Combine with ogg */
+	Q_strlcpy(filename, namewe, sizeof(filename));
+
+	/* Add the extension */
+	Q_strlcat(filename, ".", sizeof(filename));
+	Q_strlcat(filename, "ogg", sizeof(filename));
```
Comment (DanielGibson) [inline src/client/sound/sound.c:250]: it doesn't really matter, but why not just do it in one `Q_strlcat()` call with ".ogg"?

File: src/client/sound/ogg.c [hunk 4]
```diff
@@ -709,3 +710,62 @@ OGG_Shutdown(void)
 
 	ogg_started = false;
 }
+
+void
+OGG_LoadAsWav(char *filename, wavinfo_t *info, void **buffer)
```
Comment (DanielGibson) [inline src/client/sound/ogg.c:715]: I know lots of code in q2 does this, but maybe we should finally start using `const char*` for strings, at least in new code

File: src/client/sound/ogg.c (hunk 1 above)
Comment (0lvin) [reply src/client/sound/ogg.c]: I have changed it in 3cc39bb48e3b42847e6e5f6024e7c460f83ce114, looks like stb_vorbis_stream_length_in_samples has returned samples multiplied by channels.

File: src/client/sound/sound.c (hunk 2 above)
Comment (0lvin) [reply src/client/sound/sound.c]: It tried to check that originally file was wav, i have removed such check in 3cc39bb48e3b42847e6e5f6024e7c460f83ce114 as redundant.

File: src/client/sound/sound.c (hunk 3 above)
Comment (0lvin) [reply src/client/sound/sound.c]: fixed

File: src/client/sound/ogg.c (hunk 4 above)
Comment (0lvin) [reply src/client/sound/ogg.c:715]: Filename has used in `FS_LoadFile` and `FS_LoadFile` has used `char*`, i have updated chars to const where it not required to add `const char*` -> `char*` before use.

**Comment by Yamagi** [comment]
@0lvin, @DanielGibson: Are we happy with this?

**Comment by 0lvin** [comment]
@Yamagi Nothing to add from me, I have added little more code for flexible haptic feedback in #722 . I will be appreciate if someone will leave feedback how it "sounds", Code in #722 has exactly same duration and volume of haptic feedback as audio files and tried to detect attack and fade duration.

**Comment by Yamagi** [comment]
@DanielGibson  Are you okay with this? I want to merge this PR before continuing with #722.

**Comment by DanielGibson** [comment]
looks good!

**Comment by Yamagi** [comment]
Thank you!

**Pull request merged**